flax/linen/summary.py

```diff
@@ -262,13 +282,100 @@ def _tabulate_fn(*fn_args, **fn_kwargs):
   return _tabulate_fn
 
 
+def _get_flops(fn, *args, **kwargs):
+  e = jax.jit(fn).lower(*args, **kwargs)
+  cost = e.cost_analysis()
+  flops = int(cost['flops']) if 'flops' in cost else 0
+  return flops
+
+
+def _get_call_flops(c: module_lib._CallInfo, mutable) -> tuple[int, int]:
+  """Return the FLOPs of executing the call `c` in the call stack.
+
+  Does not perform actual computation / compilation / memory allocation.
+
+  Args:
+    c: ``_CallInfo``.
+    mutable: Can be bool, str, or list. Specifies which collections should be
+      treated as mutable: ``bool``: all/no collections are mutable. ``str``:
+      The name of a single mutable collection. ``list``: A list of names of
+      mutable collections.
+
+  Returns:
+    FLOPs of executing forward pass of `c`, and its VJP.
+  """
+  args = jax.tree_map(_from_value_representation, c.args)
+  kwargs = jax.tree_map(_from_value_representation, c.kwargs)
+
+  static_kwargs = {}
+  dynamic_kwargs = {}
+  for k, v in kwargs.items():
+    if isinstance(v, jax.ShapeDtypeStruct):
+      dynamic_kwargs[k] = v
+    else:
+      static_kwargs[k] = v
+
+  def init(rngs, *args, **dynamic_kwargs):
+    """`c.module.init` closed over static keyword arguments."""
+    return c.module.init(
+        {'params': rngs, 'dropout': rngs},
+        *args,
+        method=c.method,
+        mutable=mutable,
+        **dynamic_kwargs,
+        **static_kwargs,
+    )
+
+  rngs = jax.ShapeDtypeStruct((2,), jnp.uint)
```

Comment:
This wont work in general. We probably have to pass in the `rngs` given to `tabulate`.

Reply:
Good catch - fixed!